src/client-agent/main.c

```diff
@@ -56,6 +56,13 @@ int main(int argc, char **argv)
     const char *group = GROUPGLOBAL;
     const char *cfg = DEFAULTCPATH;
 
+    /* To check if is valid server address */
+    bool is_valid = false;
+    int i = 0;
+    int min_size = 1;
```

Comment:
This validates hostnames longer than 1. I'd suggest testing for longer than 0, rendering this variable unnecessary.